pandas/core/apply.py

```diff
@@ -471,8 +471,22 @@ def compute_dict_like(
 
                 keys += [key] * len(key_data)
                 results += key_data
-        else:
+        elif is_groupby:
             # key used for column selection and output
+
+            df = obj.obj
+            results, keys = [], []
+            for key, how in func.items():
+                for index in range(df.shape[1]):
+                    col = df.iloc[:, index]
+                    if col.name != key:
+                        continue
+
+                    series = obj._gotitem(key, ndim=1, subset=col)
+                    result = getattr(series, op_name)(how, **kwargs)
```

Comment:
Can you see if the following works instead:

```
subset = selected_obj[[key]]
subobj = obj._gotitem(key, ndim=2, subset=subset)
result = getattr(subobj, op_name)(how, **kwargs)
```

Reply:
Using `getattr` on DataFrame throws exception in this case
```
df = pd.DataFrame(
        [[1, 2, 3], [1, 3, 4], [2, 4, 5]],
        columns=["a", "b", "c"],
    )

    gb = df.groupby("a")
    result = gb.agg({"c": ["sum", "min", "max", "min"]})
    print(result)
```
This is due to `reconstruct_func` in `DataFrameGroupby` 
```
if not relabeling:
        if isinstance(func, list) and len(func) > len(set(func)):
            # GH 28426 will raise error if duplicated function names are used and
            # there is no reassigned name
            raise SpecificationError(
                "Function names must be unique if there is no new column names "
                "assigned"
            )
  ```